templates/test.html

```diff
@@ -0,0 +1,90 @@
+
+            <div class="playerstats-rank">
+                <div class="value">{{ data.player.rank }}</div>
+                <div class="label">
+                    <div class="world">World</div>
+                    <div class="rank">Rank</div>
+                </div>
+            </div>
+        </div>
+
+        <div class="playername">
+            <h1>{{ data.player.name }}</h1>
+        </div>
+    </div>
+
+    <div class="stats">
+        <div class="stats-header">
+            <div>Statistics</div>
+            <div>Ladder</div>
+            <div>Seasons</div>
+            <div>All</div>
+        </div>
+
+        <div class="stats-percentage">
+            <div class="title">Win rate</div>
+            <div class="value">
+                {{ data.player.sumLadderWon / (data.player.sumLadderTie + data.player.sumLadderLose) * 100 }}%
+            </div>
+            <div class="value">
+                {{ data.player.sumSeasonWon / (data.player.sumSeasonTie + data.player.sumSeasonLose) * 100 }}%
+            </div>
+            <div class="value">{{ data.player.sumWon / (data.player.sumTie + data.player.sumLose) * 100 }}%</div>
+        </div>
+
+        <div class="stats-ratio">
+            <div class="title">Matches</div>
+
+            <div class="value">
+                <span class="stats-ratio-win">{{ data.player.sumLadderWon }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-tie">{{ data.player.sumLadderTie }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-lose">{{ data.player.sumLadderLose }}</span>
+            </div>
+
+            <div class="value">
+                <span class="stats-ratio-win">{{ data.player.sumSeasonWon }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-tie">{{ data.player.sumSeasonTie }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-lose">{{ data.player.sumSeasonLose }}</span>
+            </div>
+
+            <div class="value">
+                <span class="stats-ratio-win">{{ data.player.sumWon }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-tie">{{ data.player.sumTie }}</span>&nbsp;/&nbsp;
+                <span class="stats-ratio-lose">{{ data.player.sumLose }}</span>
+            </div>
+        </div>
+    </div>
+
+    <div class="eloGraph">
+        Graph anhand letzter Spiele mit ELO Werten
+    </div>
+
+    <div class="gamelist">
+        {% for game in data.games %}
+        <div class="gamelist-item">
+            <span class="gamelist-item-vs">vs</span>
+
+            {% for opponent in game.opponents %}
+            <img class="gamelist-item-playerimg" src="{{ opponent.img }}" />
+            {% endfor %}
+
+            <span class="gamelist-item-playername">
+                {% for opponent in game.opponents %} {{ opponent.name }}&nbsp; {% endfor %}
+            </span>
+
+            <span class="gamelist-item-score-{{ game.win ? 'win' : 'lose' }}">
+                {{ game.result }}
+            </span>
```

Comment:
<data> = {
    'player': <playerdata>,
    'games': [<gamedata>],
}

<playerdata> = {
    'elo': <float>,
    'rank': <int>,
    'img': <string>,
    'name': <string>,
    'sumLadderWon': <int>,
    'sumLadderTie': <int>,
    'sumLadderLose': <int>,
    'sumSeasonWon': <int>,
    'sumSeasonTie': <int>,
    'sumSeasonLose': <int>,
    'sumWon': <int>,
    'sumTie': <int>,
    'sumLose': <int>
}


<gamedata> = {
    'opponents': [<playerdata>],
    'result': <string ??>,
    'win': <string ??>
}

Reply:
wie sollen game.result und game.win aussehen??

Reply:
das data object ist aus der sicht des spieler nach meinem template.

man könnte aber auch machen:
```
game = {
    participants: [ UserObject* ],
    points: [number]
    winner: UserObject,
    tie: boolean,
}
```

beispiel
```
game = {
    participants: [ UserA, UserB, UserC ],
    points: [15, 10, 1],
    winner: UserA, // oder user_id statt nochmal dem object
    tie: false,
}
```

ka up 2 you :>

Reply:
```
game = {
    participants: [ UserObject* ],
    points: [number],
    pointsRule: "highestWins" | "lowestWins",
}
```

Reply:
generell: MATCH als name (nicht game), das GAME ist die spielart (man muss ja wissen was gepsielt wurde :D)
```
<matchdata> = {
    'gamename': <string>,
    'participants': [<playerdata>],
    'result': [<integer>], # same order as participants
    'winner': <playerdata>,
    'datetime': <datetime>,
    'tie': <boolean>
}
```


ansonsten noch:
```
OUTPUT SPECIFICATION
<data> = {
    'player': <playerdata>,
    'games': [<gamedata>],
}

<playerdata> = {
    'name': <string>,
    'elo': <float>,
    'rank': <int>,
    'img': <string>,
    'sumLadderWon': <int>,
    'sumLadderTie': <int>,
    'sumLadderLose': <int>,
    'sumSeasonWon': <int>,
    'sumSeasonTie': <int>,
    'sumSeasonLose': <int>,
    'sumWon': <int>,
    'sumTie': <int>,
    'sumLose': <int>
}
```

Reply:
```
<matchdata> = {
    'gamename': <string>,
    'participants': [<playerdata>],
    'result': [<integer>], # same order as participants
    'winner': <playerdata>,
    'datetime': <datetime>,
    'tie': <boolean>,
    'pointrule': [<string>] # example: "lowerwins"
}
```

Reply:
1. matchdata fehlt:
- game img
- game id oder game internalName (für wiki reference nacher)

2.
```
<data> = {
    'player': <playerdata>,
    'games': [<gamedata>],
}
```
=
```
<data> = {
    'player': <playerdata>,
    'matches': [<matchdata>],
}
```
?

3.
```
<playerdata> = {
    'name': <string>,
    'elo': <float>,
    'rank': <int>,
    'img': <string>,
    'sumWon': <int>,
    'sumTie': <int>,
    'sumLose': <int>,
    'sumSeasonWon': <int>,
    'sumSeasonTie': <int>,
    'sumSeasonLose': <int>,
    'sumOffSeasonWon': <int>,
    'sumOffSeasonTie': <int>,
    'sumOffSeasonLose': <int>
}
```

Reply:
OUTPUT SPECIFICATION
```
<data> = {
    'player': <playerdata>,
    'matches': [<matchdata>],
}

<playerdata> = {
    'name': <string>,
    'elo': <float>,
    'rank': <int>,
    'img': <string>,
    'sumOffSeasonWon': <int>,
    'sumOffSeasonTie': <int>,
    'sumOffSeasonLose': <int>,
    'sumSeasonWon': <int>,
    'sumSeasonTie': <int>,
    'sumSeasonLose': <int>,
    'sumWon': <int>,
    'sumTie': <int>,
    'sumLose': <int>,
}


<matchdata> = {
    'game': <gamedata>,
    'participants': [<playerdata>],
    'result': [<integer>], # same order as participants
    'winner': <playerdata>,
    'datetime': <datetime>,
    'tie': <boolean>,
}

<gamedata> = {
    'name': <string>,
    'img': <string,
    'field_of_play': <string>,
    'field_of_play_icon': <string>,
    'pointrules': [<string>] # example: "lowerwins",
    'specific_leveling_value' = <integer>,
}
```

Reply:
perfection T_T